test/testOpenAPIFile.js

```diff
@@ -0,0 +1,300 @@
+				type: t
+			});
+			test.ok(oaf);
+
+			var jsonToParse = '{\n' +
+					'  "string1": "First line\\n\\nSecond line `with` code"\n' +
+					'}';
+			oaf.parse(jsonToParse);
+
+			var set = oaf.getTranslationSet();
+			test.ok(set);
+
+			var resources = set.getAll();
+			test.equal(resources.length, 2);
+			test.equal(resources[0].getSource(), 'First line');
+			test.equal(resources[0].getComment(), null);
+
+			test.equal(resources[1].getSource(), 'Second line <c0/> code');
+			test.equal(resources[1].getComment(), 'c0 will be replaced with the inline code `with`.');
+
+			test.done();
+		}
+	},
+
+	localizeText: {
+		testLocalizeTextSimpleJson: function(test) {
+			test.expect(2);
+
+			var oaf = new OpenAPIFile({
+				project: p,
+				type: t
+			});
+
+			test.ok(oaf);
+
+			var jsonToParse = '{\n' +
+					'    "description": "This is a test description",\n' +
+					'    "title": "this is a test title"\n' +
+					'}';
+			oaf.parse(jsonToParse);
+
+			var translations = new TranslationSet();
+			translations.add(new ResourceString({
+				project: 'foo',
+				key: oaf.makeKey('This is a test description'),
+				source: 'This is a test description',
+				sourceLocale: 'en-US',
+				target: 'Это тестовое описание',
+				targetLocale: 'ru-RU',
+				datatype: 'markdown'
+			}));
+
+			var actual = oaf.localizeText(translations, 'ru-RU');
+			var expected = '{\n' +
+					'    "description": "Это тестовое описание",\n' +
+					'    "title": "this is a test title"\n' +
+					'}\n';
+
+			test.equal(actual, expected);
+
+			test.done();
+		},
+
+		testLocalizeTextMarkdownJson: function(test) {
+			test.expect(2);
+
+			var oaf = new OpenAPIFile({
+				project: p,
+				type: t
+			});
+
+			test.ok(oaf);
+
+			var jsonToParse = '{\n' +
+					'    "description": "Description with `code` in it",\n' +
+					'    "title": "this is a test title",\n' +
+					'    "additionalField": "Non-translatable field"\n' +
+					'}';
+			oaf.parse(jsonToParse);
+
+			var translations = new TranslationSet();
+			translations.add(new ResourceString({
+				project: 'foo',
+				key: oaf.makeKey('Description with <c0/> in it'),
+				source: 'Description with <c0/> in it',
+				sourceLocale: 'en-US',
+				target: 'Описание с <c0/> внутри',
+				targetLocale: 'ru-RU',
+				datatype: 'markdown'
+			}));
+			translations.add(new ResourceString({
+				project: 'foo',
+				key: oaf.makeKey('this is a test title'),
+				source: 'this is a test title',
+				sourceLocale: 'en-US',
+				target: 'это тестовое описание',
+				targetLocale: 'ru-RU',
+				datatype: 'markdown'
+			}));
+
+			var actual = oaf.localizeText(translations, 'ru-RU');
+			var expected = '{\n' +
+					'    "description": "Описание с `code` внутри",\n' +
+					'    "title": "это тестовое описание",\n' +
+					'    "additionalField": "Non-translatable field"\n' +
+					'}\n';
+
+			test.equal(actual, expected);
+
+			test.done();
+		}
+	},
+
+	localize: {
+		testLocalize: function(test) {
+			test.expect(0);
```

Comment:
Need more tests here

Reply:
Added a test for localize method consisting of three steps:
1. Reading file and extracting strings.
2. Localizing it using mock `TranslationSet`
3. Checked the output resource file to be exactly as expected.

Also tested a couple of corner cases in this test, like string that was split into two due to a double new line and only one half has translation, resulting in half-translated string in the resource file.